postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py

```diff
@@ -0,0 +1,387 @@
+     and approximates event duration by multiplying dt with the number of timesteps,
+     where maximum on-fault slip rate is above a threshold (slipRateThreshold)"""
+    
+    slipRateThreshold=args.slipRateThreshold[0]
+    if nprocs == 1 or not args.parallelLoading:
+        ASl = faultxdmf.ReadData("ASl").T[::stepsize,timeIndicesFault[0]:timeIndicesFault[1]]
+    else:
+        ASl = LoadSingleComponentParallel("ASl", surfaceData=False)
+    ASR = np.gradient(ASl, dtFault, axis=1)
+    slippingElement = np.where(np.amax(ASR, axis=0) > slipRateThreshold, 1, 0)
+    duration = slippingElement[np.argmax(slippingElement):-np.argmax(np.flip(slippingElement))].size * dtFault
+    slippingFault = np.where(ASR > slipRateThreshold, 1, 0)
+    indHypocenter = np.argmax(slippingFault[:,np.argmax(np.amax(slippingFault, axis=0))])
+    hypocenter = faultxyz[indHypocenter,:]
+    return duration, hypocenter
+
+def PickPPhases(trigger=0.001):
+    """Picks the P-wave arrival time for each receiver. This function is only suited for synthetic data without noise"""
+    
+    data = np.sqrt(u**2 + v**2 + w**2)
+    thresholds = np.amax(data, axis=1)*trigger  # trigger dictates the portion of the maximum ground velocity,
+    thresholds = thresholds.reshape(-1,1)       # which needs to be reached to pick the p-phase
+    ind = np.argmax(data>thresholds, axis=1)
+    t = ind*dt
+    return np.transpose(np.vstack((ind,t))) # p-phase output: first column index and second column simulation time
+
+def compute_P2S_window_duration(xyz, pphase, hypocenter, centroid):
+    """Approximates the difference between P- and S-wave arrivals for each receiver"""
+    
+    if not args.avgSWaveVelocity:
+        distance1 = np.linalg.norm((xyz - hypocenter), axis=1)
+        swaveVelocity = np.mean(distance1 / pphase) / np.sqrt(3)
+    else:
+        swaveVelocity = args.avgSWaveVelocity[0]
+    print("Average S-wave velocity near the surface: "+str(swaveVelocity))
+    distance2 = np.linalg.norm((xyz - centroid), axis=1)
+    swaveDelay = distance2 / swaveVelocity * (1-1/np.sqrt(3))
+    return swaveDelay
+
+def GetTimeIndices(xdmfFile):
+    if args.events[0] == 0:
+        return [0, xdmfFile.ReadNdt()]
+    elif args.events[0] == 1:
+        return [0, int(xdmfFile.ReadNdt()/2)]
+    else:
+        return [int(xdmfFile.ReadNdt()/2), xdmfFile.ReadNdt()]
+
+start = timeit.default_timer()
+
+print("Loading data...")
+surfacexdmf = sx.seissolxdmf(args.filename)
+dt = surfacexdmf.ReadTimeStep()
+nElements = surfacexdmf.ReadNElements()
+stepsize = args.step[0]
+surfacexyz = ComputeTriangleMidpoints(surfacexdmf.ReadGeometry(), surfacexdmf.ReadConnect())[::stepsize]
+
+if stepsize != 1 and args.output != "numpy":
+    sys.exit("Xdmf output is not compatible with stepsize != 1")
+
+timeIndices = GetTimeIndices(surfacexdmf)
+
+nprocs  = args.MP[0]
+assert(nprocs<=cpu_count())
+print("Using "+str(nprocs)+" ranks")
```

Comment:
just for info, the recommended way to format a string in modern python is f strings e.g.:
print(f"Using {nproc} ranks")

Reply:
Ok, I will use it in the future